<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Graded Assessment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. 

Use the links below to read the data files using the Pandas library. 

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


Execute the below code cell to avoid prettytable default error.


In [1]:
!pip install ipython-sql prettytable

import prettytable

prettytable.DEFAULT = 'DEFAULT'

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.4/44.4 kB 5.9 MB/s eta 0:00:00


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`


In [71]:
!pip install pandas
import pandas as pd
import csv, sqlite3
conn = sqlite3.connect('FinalDB.db')
cur = conn.cursor()

Load the SQL magic module


In [72]:
!pip install ipython-sql
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


Use `Pandas` to load the data available in the links above to dataframes. Use these dataframes to load data on to the database `FinalDB.db` as required tables.


In [76]:
import pandas as pd
df1 = pd.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv")
df1.to_sql("CHICAGO_CENSUS_DATA", conn, if_exists='replace', index=False, method="multi")

df2 = pd.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv")
df2.to_sql("CHICAGO_PUBLIC_SCHOOLS", conn, if_exists='replace', index=False, method="multi")

df3 = pd.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv")
df3.to_sql("CRIME_DATA", conn, if_exists='replace', index=False, method="multi")


533

In [80]:
df1.head(10)

COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT_OF_HOUSING_CROWDED  \
0                    1.0         Rogers Park                         7.7   
1                    2.0          West Ridge                         7.8   
2                    3.0              Uptown                         3.8   
3                    4.0      Lincoln Square                         3.4   
4                    5.0        North Center                         0.3   
5                    6.0           Lake View                         1.1   
6                    7.0        Lincoln Park                         0.8   
7                    8.0     Near North Side                         1.9   
8                    9.0         Edison Park                         1.1   
9                   10.0        Norwood Park                         2.0   

   PERCENT_HOUSEHOLDS_BELOW_POVERTY  PERCENT_AGED_16__UNEMPLOYED  \
0                              23.6                          8.7   
1                              17.2                          8.8   
2                              24.0                          8.9   
3                              10.9                          8.2   
4                               7.5                          5.2   
5                              11.4                          4.7   
6                              12.3                          5.1   
7                              12.9                          7.0   
8                               3.3                          6.5   
9                               5.4                          9.0   

   PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA  \
0                                          18.2   
1                                          20.8   
2                                          11.8   
3                                          13.4   
4                                           4.5   
5                                           2.6   
6                                           3.6   
7                                           2.5   
8                                           7.4   
9                                          11.5   

   PERCENT_AGED_UNDER_18_OR_OVER_64  PER_CAPITA_INCOME  HARDSHIP_INDEX  
0                              27.5              23939            39.0  
1                              38.5              23040            46.0  
2                              22.2              35787            20.0  
3                              25.5              37524            17.0  
4                              26.2              57123             6.0  
5                              17.0              60058             5.0  
6                              21.5              71551             2.0  
7                              22.6              88669             1.0  
8                              35.3              40959             8.0  
9                              39.5              32875            21.0

In [81]:
df1.tail(10)

COMMUNITY_AREA_NUMBER     COMMUNITY_AREA_NAME  PERCENT_OF_HOUSING_CROWDED  \
68                   69.0  Greater Grand Crossing                         3.6   
69                   70.0                 Ashburn                         4.0   
70                   71.0          Auburn Gresham                         4.0   
71                   72.0                 Beverly                         0.9   
72                   73.0       Washington Height                         1.1   
73                   74.0         Mount Greenwood                         1.0   
74                   75.0             Morgan Park                         0.8   
75                   76.0                  O'Hare                         3.6   
76                   77.0               Edgewater                         4.1   
77                    NaN                 CHICAGO                         4.7   

    PERCENT_HOUSEHOLDS_BELOW_POVERTY  PERCENT_AGED_16__UNEMPLOYED  \
68                              29.6                         23.0   
69                              10.4                         11.7   
70                              27.6                         28.3   
71                               5.1                          8.0   
72                              16.9                         20.8   
73                               3.4                          8.7   
74                              13.2                         15.0   
75                              15.4                          7.1   
76                              18.2                          9.2   
77                              19.7                         12.9   

    PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA  \
68                                          16.5   
69                                          17.7   
70                                          18.5   
71                                           3.7   
72                                          13.7   
73                                           4.3   
74                                          10.8   
75                                          10.9   
76                                           9.7   
77                                          19.5   

    PERCENT_AGED_UNDER_18_OR_OVER_64  PER_CAPITA_INCOME  HARDSHIP_INDEX  
68                              41.0              17285            66.0  
69                              36.9              23482            37.0  
70                              41.9              15528            74.0  
71                              40.5              39523            12.0  
72                              42.6              19713            48.0  
73                              36.8              34381            16.0  
74                              40.3              27149            30.0  
75                              30.3              25828            24.0  
76                              23.8              33385            19.0  
77                              33.5              28202             NaN

In [82]:
df2.head(7)

School_ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   
5     610212                  Albany Park Multicultural Academy   
6     609720                Albert G Lane Technical High School   

  Elementary, Middle, or High School         Street_Address     City State  \
0                                 ES        615 W Kemper Pl  Chicago    IL   
1                                 ES  7511 S South Shore Dr  Chicago    IL   
2                                 ES     8010 S Kostner Ave  Chicago    IL   
3                                 ES     4619 S Wolcott Ave  Chicago    IL   
4                                 HS        3630 S Wells St  Chicago    IL   
5                                 MS      4929 N Sawyer Ave  Chicago    IL   
6                                 HS      2501 W Addison St  Chicago    IL   

   ZIP_Code    Phone_Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   
5     60625  (773) 534-5108   
6     60618  (773) 534-5400   

                                                Link  \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   
5  http://schoolreports.cps.edu/SchoolProgressRep...   
6  http://schoolreports.cps.edu/SchoolProgressRep...   

                            Network_Manager  ... Freshman_on_Track_Rate__  \
0              Fullerton Elementary Network  ...                      NDA   
1                 Skyway Elementary Network  ...                      NDA   
2                 Midway Elementary Network  ...                      NDA   
3               Pershing Elementary Network  ...                      NDA   
4        Southwest Side High School Network  ...                     91.8   
5                 O'Hare Elementary Network  ...                      NDA   
6  North-Northwest Side High School Network  ...                     90.7   

  X_COORDINATE Y_COORDINATE   Latitude  Longitude COMMUNITY_AREA_NUMBER  \
0  1171699.458  1915829.428  41.924497 -87.644522                     7   
1  1196129.985  1856209.466  41.760324 -87.556736                    43   
2  1148427.165  1851012.215  41.747111 -87.731702                    70   
3  1164504.290  1873959.199  41.809757 -87.672145                    61   
4  1175177.622  1880745.126  41.828146 -87.632794                    34   
5  1153858.196  1932691.891  41.971143 -87.709627                    14   
6  1158975.392  1923791.705  41.946617 -87.691056                     5   

  COMMUNITY_AREA_NAME  Ward Police_District                     Location  
0        LINCOLN PARK    43              18  (41.92449696, -87.64452163)  
1         SOUTH SHORE     7               4  (41.76032435, -87.55673627)  
2             ASHBURN    13               8  (41.74711093, -87.73170248)  
3            NEW CITY    20               9    (41.8097569, -87.6721446)  
4       ARMOUR SQUARE    11               9  (41.82814609, -87.63279369)  
5         ALBANY PARK    39              17   (41.9711433, -87.70962725)  
6        NORTH CENTER    47              19  (41.94661693, -87.69105603)  

[7 rows x 78 columns]

In [83]:
df2.tail(7)

School_ID                              NAME_OF_SCHOOL  \
559     610232        Williams Multiplex Elementary School   
560     610336  Williams Preparatory Academy Middle School   
561     610172         William T Sherman Elementary School   
562     609844          William W Carter Elementary School   
563     610088         Wolfgang A Mozart Elementary School   
564     609977        Woodlawn Community Elementary School   
565     610392          World Language Academy High School   

    Elementary, Middle, or High School       Street_Address     City State  \
559                                 ES   2710 S Dearborn St  Chicago    IL   
560                                 MS   2710 S Dearborn St  Chicago    IL   
561                                 ES       1000 W 52nd St  Chicago    IL   
562                                 ES  5740 S Michigan Ave  Chicago    IL   
563                                 ES    2200 N Hamlin Ave  Chicago    IL   
564                                 ES   6657 S Kimbark Ave  Chicago    IL   
565                                 HS   3120 S Kostner Ave  Chicago    IL   

     ZIP_Code    Phone_Number  \
559     60616  (773) 534-9245   
560     60616  (773) 534-9235   
561     60609  (773) 535-1757   
562     60637  (773) 535-0860   
563     60647  (773) 534-4160   
564     60637  (773) 535-0801   
565     60623  (773) 535-4334   

                                                  Link  \
559  http://schoolreports.cps.edu/SchoolProgressRep...   
560  http://schoolreports.cps.edu/SchoolProgressRep...   
561  http://schoolreports.cps.edu/SchoolProgressRep...   
562  http://schoolreports.cps.edu/SchoolProgressRep...   
563  http://schoolreports.cps.edu/SchoolProgressRep...   
564  http://schoolreports.cps.edu/SchoolProgressRep...   
565  http://schoolreports.cps.edu/SchoolProgressRep...   

                     Network_Manager  ... Freshman_on_Track_Rate__  \
559  Burnham Park Elementary Network  ...                      NDA   
560  Burnham Park Elementary Network  ...                      NDA   
561                     AUSL Schools  ...                      NDA   
562  Burnham Park Elementary Network  ...                      NDA   
563     Fullerton Elementary Network  ...                      NDA   
564  Burnham Park Elementary Network  ...                      NDA   
565    West Side High School Network  ...                       76   

    X_COORDINATE Y_COORDINATE   Latitude  Longitude COMMUNITY_AREA_NUMBER  \
559  1176275.784  1886545.486  41.844038 -87.628590                    35   
560  1176275.784  1886545.486  41.844038 -87.628590                    35   
561  1170500.817  1870373.159  41.799788 -87.650255                    61   
562  1178101.365  1866810.123  41.789841 -87.622490                    40   
563  1150644.396  1914368.955  41.920927 -87.721925                    22   
564  1185825.188  1860883.579  41.773400 -87.594356                    42   
565  1147521.302  1883405.128  41.836020 -87.734195                    30   

    COMMUNITY_AREA_NAME  Ward Police_District                     Location  
559             DOUGLAS     3               1  (41.84403809, -87.62859008)  
560             DOUGLAS     3               1  (41.84403809, -87.62859008)  
561            NEW CITY    16               9  (41.79978772, -87.65025483)  
562     WASHINGTON PARK    20               2  (41.78984129, -87.62248974)  
563        LOGAN SQUARE    35              25  (41.92092734, -87.72192541)  
564            WOODLAWN     5               3  (41.77339962, -87.59435584)  
565      SOUTH LAWNDALE    22              10  (41.83601953, -87.73419465)  

[7 rows x 78 columns]

In [84]:
df3.head(5)

ID CASE_NUMBER        DATE                     BLOCK IUCR  \
0   3512276    HK587712  2004-08-28        047XX S KEDZIE AVE  890   
1   3406613    HK456306  2004-06-26  009XX N CENTRAL PARK AVE  820   
2   8002131    HT233595  2011-04-04        043XX S WABASH AVE  820   
3   7903289    HT133522  2010-12-30      083XX S KINGSTON AVE  840   
4  10402076    HZ138551  2016-02-02           033XX W 66TH ST  820   

  PRIMARY_TYPE                    DESCRIPTION          LOCATION_DESCRIPTION  \
0        THEFT                  FROM BUILDING            SMALL RETAIL STORE   
1        THEFT                 $500 AND UNDER                         OTHER   
2        THEFT                 $500 AND UNDER  NURSING HOME/RETIREMENT HOME   
3        THEFT  FINANCIAL ID THEFT: OVER $300                     RESIDENCE   
4        THEFT                 $500 AND UNDER                         ALLEY   

   ARREST  DOMESTIC  ...  DISTRICT  WARD  COMMUNITY_AREA_NUMBER  FBICODE  \
0   False     False  ...         9  14.0                   58.0        6   
1   False     False  ...        11  27.0                   23.0        6   
2   False     False  ...         2   3.0                   38.0        6   
3   False     False  ...         4   7.0                   46.0        6   
4   False     False  ...         8  15.0                   66.0        6   

  X_COORDINATE  Y_COORDINATE  YEAR   LATITUDE  LONGITUDE  \
0    1155838.0     1873050.0  2004  41.807440 -87.703956   
1    1152206.0     1906127.0  2004  41.898280 -87.716406   
2    1177436.0     1876313.0  2011  41.815933 -87.624642   
3    1194622.0     1850125.0  2010  41.743665 -87.562463   
4    1155240.0     1860661.0  2016  41.773455 -87.706480   

                        LOCATION  
0    (41.8074405, -87.703955849)  
1  (41.898279962, -87.716405505)  
2  (41.815933131, -87.624642127)  
3  (41.743665322, -87.562462756)  
4  (41.773455295, -87.706480471)  

[5 rows x 21 columns]

In [86]:
df3.tail(1)

ID CASE_NUMBER        DATE                BLOCK  IUCR  \
532  1326195     G021609  2001-01-11  087XX S ESCANABA AV  9901   

          PRIMARY_TYPE        DESCRIPTION LOCATION_DESCRIPTION  ARREST  \
532  DOMESTIC VIOLENCE  DOMESTIC VIOLENCE            APARTMENT    True   

     DOMESTIC  ...  DISTRICT  WARD  COMMUNITY_AREA_NUMBER  FBICODE  \
532      True  ...         4   NaN                    NaN      08B   

    X_COORDINATE  Y_COORDINATE  YEAR   LATITUDE  LONGITUDE  \
532    1196869.0     1847416.0  2001  41.736176  -87.55432   

                         LOCATION  
532  (41.73617608, -87.554319607)  

[1 rows x 21 columns]

### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. 

Use the links below to read the data files using the Pandas library. 

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


Establish a connection between SQL magic module and the database `FinalDB.db`


In [87]:
%sql sqlite:///FinalDB.db

You can now proceed to the the following questions. Please note that a graded assignment will follow this lab and there will be a question on each of the problems stated below. It can be from the answer you received or the code you write for this problem. Therefore, please keep a note of both your codes as well as the response you generate.


## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [89]:
%sql SELECT COUNT(*) total_crimes_recorded FROM CRIME_DATA;

 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [94]:
%sql SELECT COMMUNITY_AREA_NAME, COMMUNITY_AREA_NUMBER FROM CHICAGO_CENSUS_DATA WHERE per_capita_income < 11000;

 * sqlite:///FinalDB.db
Done.


[('West Garfield Park', 26.0),
 ('South Lawndale', 30.0),
 ('Fuller Park', 37.0),
 ('Riverdale', 54.0)]

In [1]:
%sql SELECT COMMUNITY_AREA_NAME FROM CHICAGO_CENSUS_DATA WHERE per_capita_income > 50000;

UsageError: Line magic function `%sql` not found.


### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis) 


In [98]:
%sql SELECT CASE_NUMBER, DESCRIPTION FROM CRIME_DATA WHERE DESCRIPTION LIKE '%MINOR%'

 * sqlite:///FinalDB.db
Done.


[('HL266884', 'SELL/GIVE/DEL LIQUOR TO MINOR'),
 ('HK238408', 'ILLEGAL CONSUMPTION BY MINOR')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [101]:
%sql SELECT PRIMARY_TYPE, DESCRIPTION FROM CRIME_DATA WHERE PRIMARY_TYPE LIKE '%KIDNAPPING%' AND DESCRIPTION LIKE '%CHILD%'

 * sqlite:///FinalDB.db
Done.


[('KIDNAPPING', 'CHILD ABDUCTION/STRANGER')]

### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [105]:
%sql SELECT DISTINCT PRIMARY_TYPE AS CRIME_TYPE FROM CRIME_DATA WHERE LOCATION_DESCRIPTION LIKE '%SCHOOL%'

 * sqlite:///FinalDB.db
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 6

##### List the type of schools along with the average safety score for each type.


In [108]:
%sql SELECT "Elementary, Middle, or High School", SAFETY_SCORE FROM CHICAGO_PUBLIC_SCHOOLS; 

 * sqlite:///FinalDB.db
Done.


[('ES', 99.0),
 ('ES', 54.0),
 ('ES', 61.0),
 ('ES', 56.0),
 ('HS', 49.0),
 ('MS', 66.0),
 ('HS', 88.0),
 ('ES', 67.0),
 ('HS', 70.0),
 ('ES', 43.0),
 ('ES', 99.0),
 ('ES', 49.0),
 ('ES', 73.0),
 ('ES', 31.0),
 ('ES', 28.0),
 ('ES', 19.0),
 ('ES', 37.0),
 ('ES', 46.0),
 ('HS', 52.0),
 ('ES', 45.0),
 ('ES', 81.0),
 ('MS', 38.0),
 ('ES', 23.0),
 ('ES', 57.0),
 ('ES', 48.0),
 ('ES', 86.0),
 ('ES', 31.0),
 ('ES', 99.0),
 ('ES', 59.0),
 ('ES', 32.0),
 ('ES', None),
 ('ES', 33.0),
 ('ES', 27.0),
 ('ES', 36.0),
 ('ES', 47.0),
 ('ES', 87.0),
 ('ES', None),
 ('ES', 15.0),
 ('ES', 48.0),
 ('ES', 99.0),
 ('HS', 38.0),
 ('ES', 23.0),
 ('HS', 38.0),
 ('ES', 32.0),
 ('ES', None),
 ('ES', 75.0),
 ('HS', 46.0),
 ('ES', 36.0),
 ('ES', 27.0),
 ('ES', 50.0),
 ('ES', 39.0),
 ('ES', 45.0),
 ('ES', None),
 ('ES', 63.0),
 ('ES', 27.0),
 ('ES', 55.0),
 ('HS', 41.0),
 ('ES', 61.0),
 ('ES', 51.0),
 ('ES', 58.0),
 ('HS', 48.0),
 ('ES', 68.0),
 ('ES', 54.0),
 ('ES', 35.0),
 ('ES', 26.0),
 ('ES', 54.0),
 ('HS', 59.0),
 ('ES', 42.0),
 ('ES', 44.0),
 ('ES', 40.0),
 ('ES', 32.0),
 ('ES', 36.0),
 ('ES', 53.0),
 ('ES', 25.0),
 ('HS', 26.0),
 ('ES', 58.0),
 ('ES', 16.0),
 ('ES', 20.0),
 ('ES', 55.0),
 ('ES', 41.0),
 ('ES', 24.0),
 ('ES', 66.0),
 ('HS', 58.0),
 ('HS', 87.0),
 ('HS', 32.0),
 ('HS', 27.0),
 ('ES', 78.0),
 ('HS', 36.0),
 ('ES', 66.0),
 ('ES', 25.0),
 ('ES', 36.0),
 ('HS', 60.0),
 ('ES', 59.0),
 ('ES', 14.0),
 ('ES', 45.0),
 ('ES', 27.0),
 ('ES', None),
 ('ES', 52.0),
 ('ES', None),
 ('HS', 50.0),
 ('ES', 39.0),
 ('ES', 42.0),
 ('ES', 30.0),
 ('ES', 48.0),
 ('HS', 43.0),
 ('HS', None),
 ('ES', 41.0),
 ('ES', 47.0),
 ('ES', None),
 ('ES', 28.0),
 ('ES', 34.0),
 ('ES', None),
 ('HS', None),
 ('ES', 19.0),
 ('ES', 46.0),
 ('ES', 61.0),
 ('ES', 46.0),
 ('HS', 27.0),
 ('ES', 99.0),
 ('ES', 99.0),
 ('ES', 95.0),
 ('ES', 1.0),
 ('ES', 33.0),
 ('ES', 47.0),
 ('ES', 38.0),
 ('ES', 22.0),
 ('ES', None),
 ('ES', 50.0),
 ('ES', 30.0),
 ('ES', 77.0),
 ('HS', 34.0),
 ('ES', 27.0),
 ('HS', 26.0),
 ('ES', 44.0),
 ('ES', 29.0),
 ('ES', 44.0),
 ('ES', 86.0),
 ('ES', 51.0),
 ('ES', 70.0),
 ('ES', 31.0),
 ('HS', 30.0),
 ('ES', 99.0),
 ('HS', 13.0),
 ('ES', 60.0),
 ('ES', 31.0),
 ('ES', 30.0),
 ('ES', None),
 ('HS', 67.0),
 ('ES', 71.0),
 ('ES', 36.0),
 ('ES', 42.0),
 ('ES', 44.0),
 ('MS', 60.0),
 ('ES', 31.0),
 ('ES', 53.0),
 ('ES', None),
 ('ES', 51.0),
 ('ES', 29.0),
 ('ES', 20.0),
 ('ES', 35.0),
 ('ES', 34.0),
 ('ES', 11.0),
 ('MS', 49.0),
 ('ES', 23.0),
 ('ES', 38.0),
 ('ES', 37.0),
 ('ES', 30.0),
 ('ES', 46.0),
 ('ES', 27.0),
 ('ES', 87.0),
 ('ES', 46.0),
 ('ES', 48.0),
 ('ES', None),
 ('ES', 60.0),
 ('ES', 86.0),
 ('HS', 44.0),
 ('ES', 16.0),
 ('ES', None),
 ('ES', 72.0),
 ('HS', 70.0),
 ('HS', 14.0),
 ('ES', 65.0),
 ('ES', 28.0),
 ('ES', 61.0),
 ('ES', 59.0),
 ('ES', 49.0),
 ('ES', 61.0),
 ('ES', 76.0),
 ('HS', 33.0),
 ('ES', None),
 ('ES', 43.0),
 ('ES', 59.0),
 ('ES', 39.0),
 ('HS', 53.0),
 ('ES', 25.0),
 ('ES', 58.0),
 ('HS', 46.0),
 ('ES', 23.0),
 ('HS', 74.0),
 ('ES', 6.0),
 ('ES', 44.0),
 ('ES', 49.0),
 ('ES', 35.0),
 ('HS', 43.0),
 ('ES', 64.0),
 ('ES', 26.0),
 ('HS', 42.0),
 ('HS', 64.0),
 ('ES', 78.0),
 ('ES', 35.0),
 ('ES', 28.0),
 ('ES', 28.0),
 ('ES', 68.0),
 ('ES', 92.0),
 ('ES', 29.0),
 ('ES', 58.0),
 ('ES', 43.0),
 ('ES', 38.0),
 ('ES', None),
 ('ES', 41.0),
 ('ES', 33.0),
 ('ES', 45.0),
 ('HS', None),
 ('ES', 64.0),
 ('HS', 31.0),
 ('ES', 65.0),
 ('ES', 28.0),
 ('ES', 42.0),
 ('HS', 29.0),
 ('HS', 64.0),
 ('ES', 49.0),
 ('ES', 31.0),
 ('HS', 58.0),
 ('ES', 67.0),
 ('ES', 39.0),
 ('MS', 34.0),
 ('ES', 57.0),
 ('ES', 44.0),
 ('ES', 17.0),
 ('ES', None),
 ('ES', 63.0),
 ('HS', 57.0),
 ('ES', 55.0),
 ('ES', 71.0),
 ('ES', 99.0),
 ('ES', 99.0),
 ('ES', 22.0),
 ('ES', 39.0),
 ('ES', 45.0),
 ('ES', 32.0),
 ('ES', 83.0),
 ('ES', 30.0),
 ('ES', 39.0),
 ('ES', 38.0),
 ('ES', None),
 ('ES', 49.0),
 ('ES', 67.0),
 ('ES', 56.0),
 ('ES', 42.0),
 ('ES', 63.0),
 ('ES', 44.0),
 ('ES', 35.0),
 ('ES', None),
 ('ES', 38.0),
 ('ES', 75.0),
 ('ES', 60.0),
 ('ES', 35.0

In [111]:
%sql SELECT "Elementary, Middle, or High School", AVG(SAFETY_SCORE) FROM CHICAGO_PUBLIC_SCHOOLS GROUP BY "Elementary, Middle, or High School";

 * sqlite:///FinalDB.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [114]:
%sql SELECT COMMUNITY_AREA_NAME FROM CHICAGO_CENSUS_DATA WHERE PERCENT_HOUSEHOLDS_BELOW_POVERTY ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC LIMIT 5;

 * sqlite:///FinalDB.db
Done.


[('Riverdale',),
 ('Fuller Park',),
 ('Englewood',),
 ('North Lawndale',),
 ('East Garfield Park',)]

### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.


In [115]:
%sql SELECT COMMUNITY_AREA_NUMBER FROM CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY COUNT(COMMUNITY_AREA_NUMBER) DESC LIMIT 1;

 * sqlite:///FinalDB.db
Done.


[(25.0,)]

In [118]:
%sql SELECT COMMUNITY_AREA_NUMBER FROM CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY COUNT(COMMUNITY_AREA_NUMBER) DESC LIMIT 7;

 * sqlite:///FinalDB.db
Done.


[(25.0,), (23.0,), (68.0,), (29.0,), (28.0,), (8.0,), (71.0,)]


Double-click **here** for a hint

<!--
Query for the 'community area number' that has most number of incidents
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [122]:
%sql SELECT COMMUNITY_AREA_NAME, HARDSHIP_INDEX FROM CHICAGO_CENSUS_DATA WHERE HARDSHIP_INDEX = (SELECT MAX(HARDSHIP_INDEX) FROM CHICAGO_CENSUS_DATA)

 * sqlite:///FinalDB.db
Done.


[('Riverdale', 98.0)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [124]:
%sql SELECT COMMUNITY_AREA_NAME FROM CHICAGO_CENSUS_DATA WHERE COMMUNITY_AREA_NUMBER = (SELECT COMMUNITY_AREA_NUMBER FROM CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY COUNT(COMMUNITY_AREA_NUMBER) DESC LIMIT 1);

 * sqlite:///FinalDB.db
Done.


[('Austin',)]

## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>
<h4>Abhishek Gagneja</h4>
<!--
## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
|2023-10-18  | 2.6     | Abhishek Gagneja  | Modified instruction set |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |
-->
## <h3 align="center"> © IBM Corporation 2023. All rights reserved. <h3/>
